src/app/hooks/useTelemetry.ts

```diff
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { usePathname } from "next/navigation";
+import { useGa } from "./useGa";
+import { useGlean } from "./useGlean";
+
+type RecordParams = {
+  action: string;
+} & {
+  [key: string]: string;
+};
+
+export const useTelemetry = () => {
+  const path = usePathname();
+  const glean = useGlean();
+  const { gtag } = useGa();
+
+  const record = (eventName: string, params: RecordParams) => {
+    if (!glean) {
+      return;
+    }
+
+    const { action, ...otherParams } = params;
+
+    // Record event via Glean
+    glean[eventName]?.[action]?.record({ path, ...otherParams });
```

Comment:
That’s what I’m mostly unsure about: Since the generated Glean modules are being exported by `useGlean` we can record events in a more generic way than having to explicitly call them. I’m hoping to nail the types down and have a way to get them from `metrics.yaml`.